### DISCUSSION TITLE
Async  tests hang  when executed in parallel

### DISCUSSION THREAD

**Discussion by bhanu-pratap** [xunit/xunit#2305] [Question] [answered]
when i am running the set of selenium tests in parallel then the Async test hangs in the middle, however when executed in sequence it runs fine.
This is only happening with the async tests and other tests runs fine.
Not able to understand what is causing this weird behavior. Please help!

**Comment by bradwilson** [comment, marked as answer]
The most common cause of hung async tests is inappropriate use of thread-blocking `Task` APIs like `.Result`, `.Wait()`, `.GetAwaiter().GetResult()`, and `Task.WaitAll()`/`Task.WaitAny()`.

One way to verify this is to turn off thread limitations by [setting `maxParallelThreads` to `-1`](https://xunit.net/docs/configuration-files#maxParallelThreads). If this causes the hangs to go away, then you know you're creating the deadlock somewhere in your tests and/or your production code.
